### PR TITLE
Updated SFBAudioEngine

### PR DESCRIPTION
**Pull request by jamesrhurst**
I've updated SFBAudioEngine which fixed a bug where when importing a large number of audio files it would eventually fail to open any files at all.

I've also made the successfully imported notification show the number of actually imported items instead of the number of items that were attempted to be imported.

### CODE REVIEW

File: Sonora/Classes/SNRAudioPlayer.mm [hunk 1]
```diff
@@ -301,21 +284,31 @@ - (BOOL)seekToTime:(NSTimeInterval)time
...
+	if (decoder == nullptr) {
+        inputSource = nullptr;
         return NO;
     }
-	decoder->SetRenderingStartedCallback(renderingStarted, (__bridge void*)self);
-	decoder->SetRenderingFinishedCallback(renderingFinished, (__bridge void*)self);
-    decoder->SetDecodingStartedCallback(decodingStarted, (__bridge void*)self);
-    decoder->SetDecodingFinishedCallback(decodingFinished, (__bridge void*)self);
+    
+    _player->SetRenderingStartedBlock(^(const SFB::Audio::Decoder &decoder) {
```
Comment (indragiek) [inline Sonora/Classes/SNRAudioPlayer.mm]: Looks like indentation is inconsistent for this block.

File: Sonora/Classes/SNRAudioMetadata.mm [hunk 2]
```diff
@@ -425,9 +426,9 @@ - (NSData*)frontCoverArtData
 
 - (void)setFrontCoverArtData:(NSData *)frontCoverArtData
 {
-	_metadata->RemoveAttachedPicturesOfType(AttachedPicture::Type::FrontCover);
-	AttachedPicture *picture = new AttachedPicture;
-	picture->SetType(AttachedPicture::Type::FrontCover);
+    _metadata->RemoveAttachedPicturesOfType(Audio::AttachedPicture::Type::FrontCover);
```
Comment (indragiek) [inline Sonora/Classes/SNRAudioMetadata.mm]: Indentation

File: Sonora/Classes/SNRAudioMetadata.mm [hunk 3]
```diff
@@ -409,14 +410,14 @@ - (void)setReplayGainAlbumPeak:(NSNumber *)replayGainAlbumPeak
 
 - (NSData*)frontCoverArtData
 {
-	std::vector<AttachedPicture *> front = _metadata->GetAttachedPicturesOfType(AttachedPicture::Type::FrontCover);
+    std::vector<std::shared_ptr<Audio::AttachedPicture>> front = _metadata->GetAttachedPicturesOfType(Audio::AttachedPicture::Type::FrontCover);
```
Comment (indragiek) [inline Sonora/Classes/SNRAudioMetadata.mm]: Indentation

**Comment by indragiek** [comment]
Thanks for doing this! Just need to fix up the indentation in a few places. The project uses tabs, width = 4.

**Comment by jamesrhurst** [comment]
Switched to tabs instead of spaces and ran a re-indent on both of those files.

**Comment by indragiek** [comment]
@jamesrhurst :+1:

**Pull request merged**